### PR TITLE
LRSD-5177 Add replace to sxp-blueprints on site initilizer.

### PR DESCRIPTION
**Pull request by abfranca**

### CODE REVIEW

**Comment by abfranca** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer/src/main/java/com/liferay/osb/site/initializer/OSBSiteInitializerImpl.java [hunk 1]
```diff
@@ -41,6 +44,8 @@ public void addOrUpdateSXPBlueprint(
 			return;
 		}
 
+		json = _replace(json, stringUtilReplaceValues);
```
Comment (joseabelenda) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer/src/main/java/com/liferay/osb/site/initializer/OSBSiteInitializerImpl.java:47]: Inline this

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer/src/main/java/com/liferay/osb/site/initializer/OSBSiteInitializerImpl.java [hunk 2]
```diff
@@ -74,6 +79,21 @@ public void addOrUpdateSXPBlueprint(
 		}
 	}
 
+	private String _replace(
```
Comment (joseabelenda) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer/src/main/java/com/liferay/osb/site/initializer/OSBSiteInitializerImpl.java:82]: There are two private _replace methods. One in BundleSiteInitializer and this one in OSBSiteInitializerImpl.
Try to create a static method in SiteInitializerUtil to reuse it instead create another one.

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 3]
```diff
@@ -3790,6 +3790,11 @@ private void _assertSXPBlueprint1() throws Exception {
 				"TESTSXPBLUEPRINT1");
 
 		Assert.assertNotNull(sxpBlueprint);
+		Assert.assertFalse(
```
Comment (joseabelenda) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:3793]: Assert if the content was replaced by correct string.

**Review by joseabelenda — CHANGES_REQUESTED** [review]
@abfranca please read my comments.
Thx

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java (hunk 3 above)
Comment (nNilton) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:3793]: Or actually you can do the same as:

Assert.assertFalse(
			pageDefinitionString.contains(
				"[$TestObjectDefinition3#Test_Object_Entry_1$]"));

**Comment by nNilton** [comment]
Besides I'd encourage you to make the update tests for _assertAssetVocabularies and _assertAssetCategories since you are adding them in bundle-2

**Comment by nNilton** [comment]
Otherwise make just the test for bundle-1 and use the pre existing categories

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer/src/main/java/com/liferay/osb/site/initializer/OSBSiteInitializerImpl.java (hunk 1 above)
Comment (abfranca) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer/src/main/java/com/liferay/osb/site/initializer/OSBSiteInitializerImpl.java]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer/src/main/java/com/liferay/osb/site/initializer/OSBSiteInitializerImpl.java (hunk 2 above)
Comment (abfranca) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer/src/main/java/com/liferay/osb/site/initializer/OSBSiteInitializerImpl.java]: Done!

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java (hunk 3 above)
Comment (abfranca) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java]: Done!

**Comment by abfranca** [comment]
ci:test:sf

**Comment by abfranca** [comment]
ci:test:sf

**Comment by joseabelenda** [comment]
resent:https://github.com/brianchandotcom/liferay-portal/pull/152636

**Pull request closed without merging**